app/routes/admin-panel/index.js

```diff
@@ -6,15 +6,15 @@ export default class AdminPanelRoute extends Route {
   @service currentSession;
   @service session;
   @service router;
+  @service impersonation;
 
   beforeModel(transition) {
     this.session.requireAuthentication(transition, 'login');
-
-    if (!this.currentSession.showAdminFeatures) {
-      this.router.replaceWith('index');
+    if (
```

Comment:
because now we are repeating ourselves and if we want to change this it will be in many places